benchalerts/clients.py

```diff
@@ -66,9 +69,78 @@ def post(self, path: str, json: dict) -> Optional[dict]:
             return res.json()
 
 
+class GitHubAppClient(_BaseClient):
+    """A client to interact with a GitHub App.
+
+    Parameters
+    ----------
+    adapter
+        A requests adapter to mount to the requests session. If not given, one will be
+        created with a backoff retry strategy.
+
+    Environment variables
+    ---------------------
+    GITHUB_APP_ID
+        The numeric GitHub App ID you can get from its settings page.
+    GITHUB_APP_PRIVATE_KEY
+        The full contents of the private key file downloaded from the App's settings
+        page.
+    """
+
+    def __init__(self, adapter: Optional[HTTPAdapter] = None):
+        app_id = os.getenv("GITHUB_APP_ID")
+        if not app_id:
+            fatal_and_log("Environment variable GITHUB_APP_ID not found")
+
+        private_key = os.getenv("GITHUB_APP_PRIVATE_KEY")
+        if not private_key:
+            fatal_and_log("Environment variable GITHUB_APP_PRIVATE_KEY not found")
+
+        super().__init__(adapter=adapter)
+        encoded_jwt = self._encode_jwt(app_id=app_id, private_key=private_key)
+        self.session.headers = {"Authorization": f"Bearer {encoded_jwt}"}
+        self.base_url = "https://api.github.com/app"
+
+    @staticmethod
+    def _encode_jwt(app_id: str, private_key: str) -> str:
+        """Create, sign, and encode a JSON web token to use for GitHub App endpoints."""
+        payload = {
+            "iss": app_id,
+            "iat": datetime.datetime.utcnow() - datetime.timedelta(minutes=1),
+            "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=10),
```

Comment:
What happens if this hits this timeout? Not sure if/when we'll run into this, but we should probably at least have a TODO/issue to call this again and retry if you get an expired token error